solr/solrj/src/java/org/apache/solr/client/solrj/util/ClientUtils.java

```diff
@@ -82,6 +82,8 @@ public static String buildRequestUrl(
       String serverRootUrl,
       String collection)
       throws MalformedURLException {
+
+    // TODO remove getBasePath support here prior to closing SOLR-17256
```

Comment:
a nocommit; no? 
I've observed we devs have not used this practice much nowadays.  Maybe because we don't want to actually see "precommit" fail and similarly don't want the PR validations to fail for it either.  I suppose if there was a separate dedicated validation, it'd be okay.  Any way; just thinking out loud here.

Reply:
I don't want to remove this line until all of the setBasePath usages are removed (which will be in a separate PR).  That'll take at least one more PR, since I'm not bundling all of that into this PR, which is more about introducing the new APIs.

So not a "nocommit"